### PR TITLE
Replace warning component with one from govuk-components

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
## Context

(This PR is being spun out of #4318. This PR should be merged before that one)

We have a `WarningComponent`. It has the same API as the respective component in `govuk-components`. We should make this the first component we move over to using that gem.

And with that component in place, we can move the warning about only adding approved users to the ‘Add a new support user’ form.

## Changes proposed in this pull request

| Before | After |
| - | - |
| before | after |

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -496,6 +500,13 @@ GEM
     simplecov_json_formatter (0.1.2)
     skylight (5.0.1)
       activesupport (>= 5.2.0)
+    slim (4.1.0)
+      temple (>= 0.7.6, < 0.9)
+      tilt (>= 2.0.6, < 2.1)
+    slim-rails (3.2.0)
+      actionpack (>= 3.1)
+      railties (>= 3.1)
+      slim (>= 3.0, < 5.0)
```
Comment (paulrobertlloyd) [inline Gemfile.lock:509]: Bit annoying that `govuk-components` adds `slim` as a dependency – is this needed? @duncanjbrown

Comment (duncanjbrown) [reply Gemfile.lock:509]: Maybe not for us, but not our choice either

File: Gemfile.lock [hunk 2]
```diff
@@ -183,6 +183,10 @@ GEM
     geocoder (1.6.6)
     globalid (0.4.2)
       activesupport (>= 4.2.0)
+    govuk-components (0.1.0)
```
Comment (peteryates) [inline Gemfile.lock:186]: 🕰️

File: Gemfile.lock (hunk 1 above)
Comment (paulrobertlloyd) [reply Gemfile.lock:509]: Will update this PR with the correct version of this gem when https://github.com/DFE-Digital/govuk-components/pull/123 is merged.

**Review by duncanjbrown — APPROVED** [review]

**Pull request merged**